src/expr.rs

```diff
@@ -1158,6 +1179,36 @@ impl<'a> Rewrite for ControlFlow<'a> {
     }
 }
 
+fn allow_single_line_if(result: &str, block: &ast::Block) -> bool {
+    if result.contains('\n') {
+        return false;
+    }
+    if block.stmts.len() == 0 {
+        return true;
+    }
+    if block.stmts.len() == 1 {
+        return is_simple_control_flow_stmt(&block.stmts[0]);
+    }
+    false
+}
+
+fn is_simple_control_flow_stmt(stmt: &ast::Stmt) -> bool {
+    match stmt.kind {
+        ast::StmtKind::Expr(ref expr) => match expr.kind {
```

Comment:
Any reason you don't want to also support this for both [`ast::StmtKind::Expr`](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ast/enum.StmtKind.html#variant.Expr) and [`ast::StmtKind::Semi`](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ast/enum.StmtKind.html#variant.Semi)?

Reply:
I was just keeping this consistent with how let-else is working. In the `let_else.rs` tests the third statement adds a new line on `StmtKindL::Semi`.